modules/programs/difftastic.nix

```diff
@@ -0,0 +1,146 @@
+{
+  config,
+  options,
+  lib,
+  pkgs,
+  ...
+}:
+let
+  cfg = config.programs.difftastic;
+
+  inherit (lib)
+    mkEnableOption
+    mkIf
+    mkMerge
+    mkOption
+    mkPackageOption
+    types
+    ;
+in
+{
+  meta.maintainers = with lib.maintainers; [ khaneliman ];
+
+  imports = [
+    (lib.mkChangedOptionModule
+      [ "programs" "git" "difftastic" "enable" ]
+      [ "programs" "difftastic" "enable" ]
+      (
+        config:
+        builtins.trace
+          "programs.git.difftastic.enable has been moved to programs.difftastic.enable and programs.difftastic.enableGitIntegration. Please update your configuration."
+          (lib.getAttrFromPath [ "programs" "git" "difftastic" "enable" ] config)
+      )
+    )
+    (lib.mkRenamedOptionModule
+      [ "programs" "git" "difftastic" "package" ]
+      [ "programs" "difftastic" "package" ]
+    )
+    (lib.mkRenamedOptionModule
+      [ "programs" "git" "difftastic" "enableAsDifftool" ]
+      [ "programs" "difftastic" "enableAsDifftool" ]
+    )
+    (lib.mkRenamedOptionModule
+      [ "programs" "git" "difftastic" "options" ]
+      [ "programs" "difftastic" "options" ]
+    )
+  ]
+  ++ (
+    let
+      mkRenamed =
+        opt:
+        lib.mkRenamedOptionModule
+          [ "programs" "git" "difftastic" opt ]
+          [ "programs" "git" "difftastic" "options" opt ];
+    in
+    map mkRenamed [
+      "background"
+      "color"
+      "context"
+      "display"
+    ]
+  )
+  ++ [
+    (lib.mkRemovedOptionModule [ "programs" "git" "difftastic" "extraArgs" ] ''
+      'programs.git.difftastic.extraArgs' has been replaced by 'programs.git.difftastic.options'
+    '')
+  ];
+
+  options.programs.difftastic = {
+    enable = mkEnableOption "difftastic, a structural diff tool";
+
+    package = mkPackageOption pkgs "difftastic" { };
+
+    enableAsDifftool = mkEnableOption "" // {
+      description = ''
+        Enable the {command}`difftastic` syntax highlighter as a git difftool.
+        See <https://github.com/Wilfred/difftastic>.
+      '';
+    };
```

Comment:
Hmm, should there be mutually exclusive `git.enableIntegration` and `git.enableDiffTool` instead?

Perhaps we should use `<vcsName>.enableIntegration` for all of them, for more future-proof option names?

It also isn't immediately obvious to me that setting one as a pager should make it impossible to configure as a `diff-tool`.

Reply:
>It also isn't immediately obvious to me that setting one as a pager should make it impossible to configure as a diff-tool.

Good catch, not sure why i did that weird either/or.
 
> Hmm, should there be mutually exclusive git.enableIntegration and git.enableDiffTool instead?
> 
> Perhaps we should use <vcsName>.enableIntegration for all of them, for more future-proof option names?

Hmm interesting idea. I think it makes sense for the "generic" cli tools that could support more than one, but `git` specific naming should be safe with `enableGitIntegration`